controllers/capabilities/backend_controller.go

```diff
@@ -174,3 +202,71 @@ func (r *BackendReconciler) validateSpec(backendResource *capabilitiesv1beta1.Ba
 		FieldErrorList: errors,
 	}
 }
+
+func (r *BackendReconciler) removeBackend(backend *capabilitiesv1beta1.Backend) error {
+	providerAccount, err := controllerhelper.LookupProviderAccount(r.Client(), backend.Namespace, backend.Spec.ProviderAccountRef, r.Logger())
+	if err != nil {
+		return err
+	}
+
+	threescaleAPIClient, err := controllerhelper.PortaClient(providerAccount)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve all product CRs that are under the same ns as the backend CR
+	opts := k8sclient.ListOptions{
+		Namespace: backend.Namespace,
+	}
+	productCRsList := &capabilitiesv1beta1.ProductList{}
+	err = r.Client().List(context.TODO(), productCRsList, &opts)
+	if err != nil {
+		return err
+	}
+
+	// fetch CRs that belong to a tenant
+	tenantProductCRs, err := r.fetchTenantProductCRs(productCRsList, backend)
+
+	// update backendUsages for each product retrieved
+	for _, productCR := range tenantProductCRs {
+		delete(productCR.Spec.BackendUsages, backend.Spec.SystemName)
+		err = r.Client().Update(context.TODO(), &productCR)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Attempt to remove backendAPI - expect error on first attempt as the backendUsage has not been removed yet from 3scale
```

Comment:
We know that this call very likely will fail just after updating product CRs. The product reconciler did not have time to delete backend resources. 

As an idea, what if we tell the controller to retry in, let's say, 2 secs? It would be only when product CRs have been updated.  wdyt?

Reply:
hey, so I've practiced this a couple of times. 
Looks like the flow already is that the product reconciliation gets thrown into a queue before the next backed reconciler run. 
Looks like the queueing happens in order.
But saying this, I'm happy to implement time based requeue

Reply:
My proposal is to trying to avoid error in the 3scale API call. Increase probability that the 3scale API will work,

Reply:
Cool, thanks Eguzki, I've set it to 2sec.